client/components/site-selector/index.jsx

```diff
@@ -165,7 +168,63 @@ module.exports = React.createClass( {
 			);
 		}
 
-		return siteElements;
+		if ( ! siteElements.length ) {
+			return <div className="site-selector__no-results">{ this.translate( 'No sites found' ) }</div>;
+		}
+
+		if ( this.props.groups && ! this.state.search ) {
+			return (
+				<div>
+					<span className="site-selector__heading">
+						{ this.translate( 'All Sites' ) }
+						<Count count={ user.get().visible_site_count } />
+					</span>
+					{ siteElements }
+				</div>
+			);
+		} else {
+			return siteElements;
+		}
+	},
+
+	renderRecentSites: function() {
+		const sites = this.props.sites.getRecentlySelected();
+
+		if ( ! sites || this.state.search || ! this.props.groups ) {
+			return null;
+		}
+
+		const recentSites = sites.map( function( site ) {
+			var siteHref;
+
+			if ( this.props.siteBasePath ) {
+				siteHref = this.getSiteBasePath( site ) + '/' + site.slug;
+			}
+
+			const isSelected = this.isSelected( site );
+
+			if ( isSelected && this.props.hideSelected ) {
+				return;
+			}
+
+			return (
+				<Site
+					site={ site }
+					href={ this.props.siteBasePath ? siteHref : null }
```

Comment:
This could just be `siteHref` as you're already doing the conditional work above